File: 2015/01/rust/src/main.rs
const INPUT_FILE:&'static str = "../input/input.txt";

fn main() -> Result<(), std::io::Error> {
    let mut floor = 0;
    let mut pos = 0;
    let mut count = 0;

    // --- Read and parse the input file ---
    for c in std::fs::read_to_string(INPUT_FILE)?.chars() {
        count += 1;
        match c {
            ')' => floor -= 1,
            '(' => floor += 1,
            _   => ()
        }
        if pos == 0 && floor == -1 { pos = count; }
    }

    // --- Puzzle 1 ---
    println!("1. Floor: {}", floor);

    // --- Puzzle 2 ---
    println!("2. Position: {}", pos);
    Ok(())
}

File: 2015/01/go/main.go
package main

import (
	"fmt"
	"log"
	"os"
	"strings"
)

// --- Constants ---
const FILENAME = "../input/input.txt"

func main() {
	// --- Read input file ---
	bytes, err := os.ReadFile(FILENAME)
	if err != nil {
		log.Fatal(err)
	}
	text := strings.TrimSpace(string(bytes))

	floor := 0
	first := 0
	for i, c := range text {
		// --- Puzzle 1 ---
		if c == '(' {
			floor++
		} else {
			floor--
		}

		// --- Puzzle 2 ---
		if first == 0 && floor == -1 {
			first = i + 1
		}
	}

	// --- Results ---
	fmt.Println("1. Floor:", floor)
	fmt.Println("2. Position:", first)
}
